main/src/main/scala/sbt/Defaults.scala

```diff
@@ -173,7 +176,33 @@ object Defaults extends BuildCommon {
       apiMappings := Map.empty,
       autoScalaLibrary :== true,
       managedScalaInstance :== true,
-      classpathEntryDefinesClass :== FileValueCache(Locate.definesClass _).get,
+      classpathEntryDefinesClass := {
+        val converter = fileConverter.value
+        val f = FileValueCache({ x: NioPath =>
+          Locate.definesClass(converter.toVirtualFile(x))
+        }).get;
+        { (x: File) =>
+          f(x.toPath)
+        }
+      },
+      allowMachinePath :== true,
+      rootPaths := {
+        val app = appConfiguration.value
+        val base = app.baseDirectory
+        val boot = app.provider.scalaProvider.launcher.bootDirectory
+        val ih = app.provider.scalaProvider.launcher.ivyHome
+        val coursierCache = csrCacheDirectory.value
+        val javaHome = Paths.get(sys.props("java.home"))
```

Comment:
Does that still work in java9+ where the jdk classes come from `jrt:/` ?

Reply:
Sort of but not elegantly. We ignore `rt.jar`-derived invalidation, so that seems to work at least for now. JAVA_HOME was still needed because there are other JARs that needs remapping.